orttraining/orttraining/core/framework/module_gradient_graph_builder.cc

```diff
@@ -188,42 +191,42 @@ void ModuleGradientGraphBuilder::AddYieldOp() {
     }
   }
 
+  // YieldOps RequiredGrad attribute specifies the indices of the required gradients.
+  ONNX_NAMESPACE::AttributeProto required_grad;
+  const std::string attribute_name = "RequiredGrad";
+  required_grad.set_name(attribute_name);
+  required_grad.set_type(ONNX_NAMESPACE::AttributeProto::INTS);
+
   // Yield inputs include all user outputs, those require output gradients come first, so Yield Op can use their shapes
   // to infer Op output shapes.
   std::vector<std::string> user_output_names_require_grad;
   std::vector<std::string> user_output_names_no_grad;
```

Comment:
These two are no longer needed. We can remove it and also the emplace_back call for these two vectors below.